File: client/src/lib/types/document.ts
export interface DocumentMeta {
	id: string;
	name: string;
	origin: 'dataset' | 'upload';
	processed: boolean;
}

export interface Paragraph {
	id: string;
	documentId: string;
	page: number;
	paragraph_enum: number;
	text: string;
	bbox: [number, number, number, number]; // pdf.js coords
	relationsCount: number;
}

export interface ParagraphRelation {
	source: string;
	target: string;
	type: 'reference' | 'semantic_similarity';
	score?: number;
}

File: server/schemas/document.py
from pydantic import BaseModel
from typing import Literal, List, Optional

class DatasetDocument(BaseModel):
    id: str
    name: str
    origin: Literal['dataset', 'upload']
    processed: bool

class Paragraph(BaseModel):
    id: str
    documentId: str
    page: int
    paragraph_enum: int
    text: str
    bbox: List[float]  # [x0, y0, x1, y1]
    relationsCount: int = 0
